### PR TITLE
Create smart decorator #60

### LINKED ISSUE #60: Create smart decorator for Response RespSmart($resp)->json($data);
```php
return new RespSmart($resp)->json($data);
```

### PR DESCRIPTION
**Pull request by klepatskiy**
close #60

### CODE REVIEW

File: src/Response/RespJson.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?php
+namespace Lipid\Response;
+
+use Lipid\Response;
+
+final class RespJson implements Response
```
Comment (agorlov) [inline src/Response/RespJson.php:16]: phpdoc

File: src/Response/RespJson.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+{
+    private $json;
+    private $orig;
+
+    public function __construct($json, Response $orig = null)
```
Comment (agorlov) [inline src/Response/RespJson.php:27]: phpdoc

File: src/Response/RespJson.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    public function withHeaders(array $headers): Response
+    {
+        return $this->orig->withHeaders($headers);
+    }
+
+    public function print(): void
+    {
+        $this->withBody(json_encode($this->json, JSON_PRETTY_PRINT | JSON_UNESCAPED_UNICODE))
+            ->withHeaders(['Content-type' => 'json'])
```
Comment (agorlov) [inline src/Response/RespJson.php]: application/json

**Review by agorlov — CHANGES_REQUESTED** [review]

**Review by agorlov — APPROVED** [review]

**Comment by 0pdd** [comment]
@klepatskiy the puzzle [#82](https://github.com/agorlov/lipid/issues/82) is still not solved.

**Pull request merged**